platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/MyLocationView.java

```diff
@@ -423,22 +421,18 @@ public void onRestoreInstanceState(Parcelable state) {
   private void toggleGps(boolean enableGps) {
     LocationEngine locationEngine = LocationSource.getLocationEngine(getContext());
     if (enableGps) {
-      // Set an initial location if one available
-      Location lastLocation = locationEngine.getLastLocation();
```

Comment:
@tobrun why not request a manual update? This way we could have a faster location reading before the `LocationEngineListener` receives the first update?